In [4]:
import numpy as np
import pandas as pd
data=pd.read_csv('Advertising.csv')

Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]

# 1. Explain the dataset

In [5]:
#reading the dataset
data

Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]

In [ ]:
1.there are 200 rows and 6 columns including sl.no
2.main data set columns are with name TV, Radio, Newspaper and Sales
2.TV,Radio and Newspaper columns indicate advertisement expenditure

# 2. Check the summary statistics and discuss the max, min, avg, median, and percentiles.


In [8]:
data.describe()

Unnamed: 0          TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000  200.000000
mean   100.500000  147.042500   23.264000   30.554000   14.022500
std     57.879185   85.854236   14.846809   21.778621    5.217457
min      1.000000    0.700000    0.000000    0.300000    1.600000
25%     50.750000   74.375000    9.975000   12.750000   10.375000
50%    100.500000  149.750000   22.900000   25.750000   12.900000
75%    150.250000  218.825000   36.525000   45.100000   17.400000
max    200.000000  296.400000   49.600000  114.000000   27.000000

In [17]:
#finding the median
print(data.median())

Unnamed: 0    100.50
TV            149.75
Radio          22.90
Newspaper      25.75
Sales          12.90
dtype: float64


# 3. Whether the same amount was spent for the three advertisements TV , Radio and Newspaper

In [ ]:
#Let the hypothesis are

H0= Same amount spent for the three advertisements,µ1=µ2=µ3
Ha= Amount spent for the advertisements are not same , μ1,µ2 and µ3 are not equal


In [ ]:
# significance level

significance level, α=0.05 

In [19]:
import numpy as np
from scipy.stats import f_oneway
x=np.array(data['TV'])
y=np.array(data['Radio'])
z=np.array(data['Newspaper'])
f_oneway(x,y,z)

F_onewayResult(statistic=358.8514595342597, pvalue=4.552931539744962e-103)

In [ ]:
# comparing the p value with significance level 0.05

p value: 4.552931539744962e-103 ≈ 4.55 x 10^-103 < 0.05

so that Rejecting the null hypothesis
ie,the Amount spent for the advertisements are not same , μ1,µ2 and µ3 are not equal.